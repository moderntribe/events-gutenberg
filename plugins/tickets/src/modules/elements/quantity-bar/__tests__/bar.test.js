/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import Bar from './../bar';

describe( '<Bar />', () => {
	test( 'render null on default values', () => {
		const component = renderer.create( <Bar /> );
		expect( component.toJSON() ).toBe( null );
	} );

	test( 'render null on zero values', () => {
		const component = renderer.create( <Bar total={ 0 } value={ 0 } /> );
		expect( component.toJSON() ).toBe( null );
	} );

	test( 'render percentage as inline style', () => {
		const component = shallow( <Bar value={ 33 } total={ 100 } /> );
		const span = component.get( 0 );
		const { style } = span.props;
		expect( style ).toEqual( { width: '33%' } );
	} );

	test( 'render percentage with custom class name', () => {
		const component = renderer.create( <Bar value={ 45 } total={ 100 } className="jest-test" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
