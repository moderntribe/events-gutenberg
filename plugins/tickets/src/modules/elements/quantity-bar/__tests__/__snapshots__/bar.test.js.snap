// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> render percentage with custom class name 1`] = `
<span
  className="jest-test"
  style={
    Object {
      "width": "45%",
    }
  }
/>
`;
